pyfar/dsp/dsp.py

```diff
@@ -1248,54 +1255,65 @@ def find_impulse_response_delay(impulse_response, N=1):
     n = int(np.ceil((N+2)/2))
 
     start_samples = np.zeros(impulse_response.cshape)
+    modes = ['real', 'complex'] if impulse_response.complex else ['real']
+    start_sample = np.zeros((len(modes), 1), dtype=float)
+
     for ch in np.ndindex(impulse_response.cshape):
         # Calculate the correlation between the impulse response and its
         # minimum phase equivalent. This requires a minimum phase equivalent
         # in the strict sense, instead of the appriximation implemented in
         # pyfar.
         n_samples = impulse_response.n_samples
+        for idx, mode in zip(range(0, len(modes)), modes):
```

Comment:
This looks better
```suggestion
        for idx, mode in enumerate(modes):
```

Reply:
thanks, your're right.

Reply:
I think we need all these changes, see the discussion in slack:
We want to calculate the delay independently for real and imaginary part, and return the minimum of both. I don't see a more elegant way.